File: client-app/src/app/types/gallery.ts
export interface Gallery {
    id: string;
    publicId: string;
    url: string;
    caption: string;
    height: number;
    width: number;
}

export class Gallery implements Gallery {
    id = "";
    publicId = "";
    url = "";
    caption = "";
    height = 0;
    width = 0;
    constructor (gallery?: Gallery) {
        if (gallery) {
            Object.assign(this, gallery);
        }
    }
}

File: API/Entities/Gallery.cs
namespace API.Entities;

public class Gallery
{
    public Guid Id { get; set; }
    public string PublicId { get; set; }
    public string Url { get; set; }
    public string Caption { get; set; }
    public int Width { get; set; }
    public int Height { get; set; }
}
